### ISSUE TITLE
Initialization of kubeclient fails when running in IPv6 single stack cluster

### ISSUE THREAD

**Issue by frauniki** [fluent/fluentd-kubernetes-daemonset#1546]
### Describe the bug

Initialization of kubeclient fails when running in IPv6 single stack cluster.

Here I am trying to fill `kubernetes_url` from ENV, but I get an error because the ipv6 address url needs to be escape by `[]`.
https://github.com/fluent/fluentd-kubernetes-daemonset/blob/master/templates/conf/kubernetes.conf.erb#L31

Change `kubernetes_url` to the following or remove it from the config and it will work correctly.
```
kubernetes_url "#{ENV['FLUENT_FILTER_KUBERNETES_URL'] || 'https://[' + ENV.fetch('KUBERNETES_SERVICE_HOST') + ']:' + ENV.fetch('KUBERNETES_SERVICE_PORT') + '/api'}"
```

### To Reproduce

Just deploy this container image to an IPv6 single stack kubernetes cluster and it should reproduce.

### Expected behavior

fluentd pod starts successfully.

### Your Environment

```markdown
- Tag of using fluentd-kubernetes-daemonset: fluent/fluentd-kubernetes-daemonset:v1.17.1-debian-s3-1.2
```

### Your Configuration

```apache
None (default config)
```

### Your Error Log

```shell
{"time":"2024-12-10 06:13:56 +0000","level":"info","message":"gem 'fluentd' version '1.16.6'"}
{"time":"2024-12-10 06:13:57 +0000","level":"info","message":"adding rewrite_tag_filter rule: message [#<Fluent::PluginHelper::RecordAccessor::Accessor:0x00007fcea304da48 @keys=\"message\">, /\\\"tag\\\":\\\"(\\w+)\\\"/, \"\", \"categorized.${tag_parts[1]}.${tag_parts[2]}.${tag_parts[3]}.${tag_parts[4]}.$1\", nil]"}
/usr/local/lib/ruby/3.2.0/uri/rfc3986_parser.rb:66:in `split': bad URI(is not URI?): "https://fdad:c5a9:94e1::1:443/api" (URI::InvalidURIError)
    from /usr/local/lib/ruby/3.2.0/uri/rfc3986_parser.rb:71:in `parse'
    from /usr/local/lib/ruby/3.2.0/uri/common.rb:193:in `parse'
    from /fluentd/vendor/bundle/ruby/3.2.0/gems/kubeclient-4.12.0/lib/kubeclient/common.rb:201:in `handle_uri'
    from /fluentd/vendor/bundle/ruby/3.2.0/gems/kubeclient-4.12.0/lib/kubeclient/common.rb:74:in `initialize_client'
    from /fluentd/vendor/bundle/ruby/3.2.0/gems/kubeclient-4.12.0/lib/kubeclient.rb:27:in `initialize'
    from /fluentd/vendor/bundle/ruby/3.2.0/gems/fluent-plugin-kubernetes_metadata_filter-3.5.2/lib/fluent/plugin/filter_kubernetes_metadata.rb:305:in `new'
    from /fluentd/vendor/bundle/ruby/3.2.0/gems/fluent-plugin-kubernetes_metadata_filter-3.5.2/lib/fluent/plugin/filter_kubernetes_metadata.rb:305:in `create_client'
    from /fluentd/vendor/bundle/ruby/3.2.0/gems/fluent-plugin-kubernetes_metadata_filter-3.5.2/lib/fluent/plugin/filter_kubernetes_metadata.rb:267:in `configure'
    from /fluentd/vendor/bundle/ruby/3.2.0/gems/fluentd-1.17.1/lib/fluent/plugin.rb:187:in `configure'
    from /fluentd/vendor/bundle/ruby/3.2.0/gems/fluentd-1.17.1/lib/fluent/agent.rb:154:in `add_filter'
    from /fluentd/vendor/bundle/ruby/3.2.0/gems/fluentd-1.17.1/lib/fluent/agent.rb:72:in `block in configure'
    from /fluentd/vendor/bundle/ruby/3.2.0/gems/fluentd-1.17.1/lib/fluent/agent.rb:64:in `each'
    from /fluentd/vendor/bundle/ruby/3.2.0/gems/fluentd-1.17.1/lib/fluent/agent.rb:64:in `configure'
    from /fluentd/vendor/bundle/ruby/3.2.0/gems/fluentd-1.17.1/lib/fluent/root_agent.rb:149:in `configure'
    from /fluentd/vendor/bundle/ruby/3.2.0/gems/fluentd-1.17.1/lib/fluent/engine.rb:105:in `configure'
    from /fluentd/vendor/bundle/ruby/3.2.0/gems/fluentd-1.17.1/lib/fluent/engine.rb:80:in `run_configure'
    from /fluentd/vendor/bundle/ruby/3.2.0/gems/fluentd-1.17.1/lib/fluent/supervisor.rb:571:in `run_supervisor'
    from /fluentd/vendor/bundle/ruby/3.2.0/gems/fluentd-1.17.1/lib/fluent/command/fluentd.rb:352:in `<top (required)>'
    from <internal:/usr/local/lib/ruby/3.2.0/rubygems/core_ext/kernel_require.rb>:86:in `require'
    from <internal:/usr/local/lib/ruby/3.2.0/rubygems/core_ext/kernel_require.rb>:86:in `require'
    from /fluentd/vendor/bundle/ruby/3.2.0/gems/fluentd-1.17.1/bin/fluentd:15:in `<top (required)>'
    from /fluentd/vendor/bundle/ruby/3.2.0/bin/fluentd:25:in `load'
    from /fluentd/vendor/bundle/ruby/3.2.0/bin/fluentd:25:in `<main>'
```

**Comment by kenhys** [comment]
We need to consider dual stack case, but anyway, this issue should be fixed.

**Comment by kenhys** [comment]
@frauniki 

Fixed versions were deployed now. Thanks!

NOTE: Because of [current support policy](https://github.com/fluent/fluentd/blob/master/SECURITY.md), update will not be shipped in v1.17 series. 

Could you use v1.16.6-debian-s3-1.3 or v1.18.0-debian-s3-1.0 images if you need it.

**Comment by frauniki** [comment]
> @frauniki
> 
> Fixed versions were deployed now. Thanks!
> 
> NOTE: Because of [current support policy](https://github.com/fluent/fluentd/blob/master/SECURITY.md), update will not be shipped in v1.17 series.
> 
> Could you use v1.16.6-debian-s3-1.3 or v1.18.0-debian-s3-1.0 images if you need it.

Thanks :)
